runtests.py

```diff
@@ -629,7 +629,8 @@ def build_extension(self, ext):
 
 
 class ErrorWriter(object):
-    match_error = re.compile(r'(warning:)?(?:.*:)?\s*([-0-9]+)\s*:\s*([-0-9]+)\s*:\s*(.*)').match
+    match_error = re.compile(
+        r'(warning:)?(performance hint:)?(?:.*:)?\s*([-0-9]+)\s*:\s*([-0-9]+)\s*:\s*(.*)').match
```

Comment:
I think we can merge the first two groups and read a general "kind" from them.
```suggestion
        r'(warning:|performance hint:)?(?:.*:)?\s*([-0-9]+)\s*:\s*([-0-9]+)\s*:\s*(.*)').match
```